app/controllers/todos_controller.rb

```diff
@@ -16,7 +19,9 @@ def new
 
   def create
     @todo = Todo.new(todo_params)
+    @todo.user = current_user
     @todo.duedate = Time.new(params[:year], params[:month], params[:day])
+    @todo.category = Category.find_by(name: params[:category][:name])

```

Comment:
I wonder if it might be simpler for the `category` params to come in as ids? Then you can set `todo.category_id`, which seems to save an extra call to the db.